examples/research_projects/vqgan-clip/img_processing.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+import PIL
+import torch
+import torchvision.transforms as T
+import torchvision.transforms.functional as TF
+from PIL import Image
+
+
+def preprocess(img, target_image_size=256):
+    s = min(img.size)
+
+    if s < target_image_size:
+        raise ValueError(f"min dim for image {s} < {target_image_size}")
+
+    r = target_image_size / s
+    s = (round(r * img.size[1]), round(r * img.size[0]))
+    img = TF.resize(img, s, interpolation=PIL.Image.LANCZOS)
+    img = TF.center_crop(img, output_size=2 * [target_image_size])
+    img = torch.unsqueeze(T.ToTensor()(img), 0)
+    return img
+
+
+def preprocess_vqgan(x):
+    x = 2.0 * x - 1.0
+    return x
+
+
+def custom_to_pil(x, process=True, mode="RGB"):
+    x = x.detach().cpu()
+    if process:
+        x = torch.clamp(x, -1.0, 1.0)
+        x = (x + 1.0) / 2.0
+    x = x.permute(1, 2, 0).numpy()
+    if process:
+        x = (255 * x).astype(np.uint8)
+    x = Image.fromarray(x)
+    if not x.mode == mode:
+        x = x.convert(mode)
+    return x
+
+
+def get_pil(x):
```

Comment:
nit: function name is a bit misleading as it doesn't return a PIL image. The logic is also repeated in `custom_to_pil`.